src/core/java/com/enderio/core/common/blockentity/EnderBlockEntity.java

```diff
@@ -76,6 +78,24 @@ public void clientTick() {
 
     }
 
+    /**
+     * Perform tick on both client and server, on the end.
+     */
+    public void endTick() {
+        if (this.level != null) {
```

Comment:
1. "!="?
2. Why does serverTick null-check level and clientTick simple NPEs? Shouldn't that be either all three or none?
3. Why does serverTick check isClientside? Any more random callers?
4. Why are the xxxTick()s not protected? From just this bit of code, I'd guess calling them from anywhere but the static tick() dispatcher would be wrong.

Reply:
Level can be null on the server when everything is loading, so we need this protection.

We should use our annotation so the serverTick can not be called on the client, so the clientlevel check is not needed.

Yes these can be protected, there should be no other points where these need to be called.

Reply:
PS: regarding point 1, I should have commented one line below. Why `return;` when the level is NOT null.

Reply:
Agree with all points - thanks Henry

Reply:
Ha right, that's an oversight on my part (but it does kind of show we may need a separate update redstone call, as it didn't change any behavior since we don't rely on it). I'll fix them up and push again.